powerline/segments/common/bat.py

```diff
@@ -189,6 +174,8 @@ def battery(pl, format='{capacity:3.0%}', steps=5, gamify=False, full_heart='O',
 		another gradient level and highlighting group, so it is OK for it to be 
 		the same as full_heart as long as necessary highlighting groups are 
 		defined.
+	:param str charging:
+		Indication of "AC charging"
```

Comment:
No, you should move `charging` to format in place of prepending it. So, two symbols: `online` and `offline` and another keyword argument to `format` string (`ac_state`: `format='{ac_state}{capacity:3.0%}'` will form a segment which never changes width: used as `format.format(ac_state=(online if ac_powered else offline), capacity=(capacity / 100.0))`; default for `offline` is simply a space).